src/test/java/uk/gov/pay/directdebit/payments/services/PaymentQueryServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.pay.directdebit.payments.services;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.directdebit.gatewayaccounts.model.GoCardlessOrganisationId;
+import uk.gov.pay.directdebit.payments.dao.PaymentDao;
+import uk.gov.pay.directdebit.payments.exception.PaymentNotFoundException;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentId;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentIdAndOrganisationId;
+import uk.gov.pay.directdebit.payments.model.Payment;
+import uk.gov.pay.directdebit.payments.model.PaymentLookupKey;
+import uk.gov.pay.directdebit.payments.model.SandboxPaymentId;
+
+import java.util.Optional;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.ignoreStubs;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.GOCARDLESS;
+import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.SANDBOX;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PaymentQueryServiceTest {
+
+    private static final SandboxPaymentId SANDBOX_PAYMENT_ID = SandboxPaymentId.valueOf("Sandy");
+    private static final GoCardlessPaymentId GOCARDLESS_PAYMENT_ID = GoCardlessPaymentId.valueOf("PM123");
+    private static final GoCardlessOrganisationId GOCARDLESS_ORGANISATION_ID = GoCardlessOrganisationId.valueOf("OR123");
+    private static final GoCardlessPaymentIdAndOrganisationId GOCARDLESS_PAYMENT_ID_AND_ORGANISATION_ID =
+            new GoCardlessPaymentIdAndOrganisationId(GOCARDLESS_PAYMENT_ID, GOCARDLESS_ORGANISATION_ID);
+
+    @Mock
+    private PaymentDao mockPaymentDao;
+
+    @Mock
+    private Payment mockPayment;
+
+    private PaymentQueryService paymentQueryService;
+
+    @Before
+    public void setUp() {
+        paymentQueryService = new PaymentQueryService(mockPaymentDao);
+    }
+
+    @Test
+    public void findByProviderIdWithSandboxPaymentIdReturnsPayment() {
+        given(mockPaymentDao.findPaymentByProviderId(SANDBOX, SANDBOX_PAYMENT_ID)).willReturn(Optional.of(mockPayment));
+
+        Payment result = paymentQueryService.findByProviderPaymentId(SANDBOX, SANDBOX_PAYMENT_ID);
+
+        assertThat(result, is(mockPayment));
+
+        verifyZeroInteractions(ignoreStubs(mockPaymentDao));
```

Comment:
Is this just verifying that no other method on the `mockPaymentDao` is being invoked besides the condition we set-up in the first line of this method?

Reply:
Yes.